### PR TITLE
Back end setup

### PR DESCRIPTION
**Pull request by ushift-d**
- API tests
- Seed Database

### CODE REVIEW

File: database/index.js [hunk 1]
```diff
@@ -16,27 +17,20 @@ db.once('open', () => {
 
 const product = mongoose.Schema({
   product_Id: Number,
```
Comment (TandyN) [inline database/index.js]: Maybe have a lowercase 'id' so `product_id: Number,`? To keep things consistent

File: database/index.js [hunk 2]
```diff
@@ -16,27 +17,20 @@ db.once('open', () => {
...
   specs: {
-    part_Number: [String],
+    part_Number: String,
     GTIN: Number,
   },
 
 });
 
-module.exports = mongoose.model('Product', product);
+module.exports.Product = mongoose.model('Product', product);
```
Comment (TandyN) [inline database/index.js]: Put this entire model in a separate file?

File: seedScript.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+const brandName = [
+  'Lorem', 'ipsum', 'dolor', 'sit', 'amet', 'consectetur', 'adipiscing', 'elit'];
+
+const categoryName = ['Lorem', 'ipsum', 'dolor', 'sit', 'amet'];
+
+const playerCount = ['1 Player', '2 Player', '3 Player', '4 Player', '5 Player', '6 Player'];
+const partNumber = ['Lorem', 'ipsum', 'dolor', 'sit', 'amet', 'consectetur', 'adipiscing', 'elit', 'Sed', 'volutpat', 'feugiat', 'consequat', 'Donec', 'id', 'faucibus', 'ligula', 'Maecenas', 'lobortis', 'laoreet', 'maximus', 'Donec', 'vel', 'nunc', 'mi', 'Nam', 'rutrum', 'rutrum', 'convallis', 'Donec', 'tempus', 'gravida', 'risus', 'laoreet', 'elementum', 'Fusce', 'congue', 'rhoncus', 'sollicitudin', 'Nunc', 'quis'];
```
Comment (TandyN) [inline seedScript.js:23]: Make the partNumber actual numbers?

File: seedScript.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+function rand(min, max) {
```
Comment (TandyN) [inline seedScript.js]: Better function name.. like `randNum` ? Can also use an arrow function. This includes for the rest of the page.

File: seedScript.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  const randomNum = Math.random() * (max - min) + min;
+  return Math.floor(randomNum);
+}
+function gtinStr(num) {
+  let gtStr = '';
+  while (gtStr.length < num) {
+    gtStr = (rand(0, 9).toString()) + gtStr;
```
Comment (TandyN) [inline seedScript.js]: The use of `+=` here?

File: seedScript.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    });
+  });
+}
+
+const seed = async () => {
+  await Product.deleteMany({});
+  await createNewProducts();
+  await insertProducts();
+};
```
Comment (TandyN) [inline seedScript.js]: Is there a reason why you just can't use mongoose to create these models and push the promises into an array? Then use Promise.all instead when you are seeding? That would remove the need of having the insertProducts function.

File: test/index.test.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    });
+    test('should send error if path does not exist', async () => {
+      const req = mockRequest('/producer');
+      const res = mockResponse();
+      await route (req, res, () => {});
+
+      expect(req.session.data).toBeUndefined();
+    });
+  });
+  });
```
Comment (TandyN) [inline test/index.test.js:29]: I'm not sure if this test is necessary to have in master?

File: server/index.js [hunk 8]
```diff
@@ -5,6 +5,9 @@ const PORT = process.env.PORT || 3004;
 const app = express();
 const client = path.join(__dirname, '/../client/dist');
 app.use(express.static(client));
+app.get('/', (req, res) => {
+  res.send('Working Path');
+});
 app.listen(PORT, () => {
   console.log(`listening on port ${PORT}!`);
 });
```
Comment (TandyN) [inline server/index.js:13]: Everything is so close together?

**Review by TandyN — CHANGES_REQUESTED** [review]
You've...
- Created a Product model
- Created a seeding script that seeds 100 products from id 1 - 100
- Added 1 route

I'm not sure if you need the test created for that route you've put in as I do not see the relevance in what the route is even doing at the moment. Unless you intend your server to send `'Working Path'` in production/master branch.

File: server/index.js (hunk 8 above)
Comment (ushift-d) [reply server/index.js:13]: Yah esLint told me I had to.

File: test/index.test.js (hunk 7 above)
Comment (ushift-d) [reply test/index.test.js:29]: This file isn't real, I have an update on another branch using axios testing my api routes and creating a mock server

File: seedScript.js (hunk 3 above)
Comment (ushift-d) [reply seedScript.js:23]: They aren't just number they're letters and numbers mixed

File: seedScript.js (hunk 6 above)
Comment (ushift-d) [reply seedScript.js]: Just preference, I don't like using promises, but I'll look into refactoring this later down the line.

File: database/models/product.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    name: String,
+    age: String,
+    player_Count: String,
+  },
+
+  specs: {
+    part_Number: String,
+    GTIN: Number,
+  },
+
```
Comment (TandyN) [inline database/models/product.js:18]: Approved after getting rid of the white space.

**Review by TandyN — COMMENTED** [review]
I see that you've moved the product model to a different file & corrected a function name.

File: server/index.js (hunk 8 above)
Comment (TandyN) [reply server/index.js:13]: That is not what ESLint is telling me. May want to look at your config if you don't like this.

**Review by TandyN — APPROVED** [review]
Please remove the white space in database/models/products.js in your model before merging.

**Pull request merged**